packages/nx/src/tasks-runner/jest-worker-task-runner.ts

```diff
@@ -0,0 +1,173 @@
+import { readFileSync } from 'fs';
+import { ChildProcess } from 'child_process';
+import { workspaceRoot } from '../utils/workspace-root';
+import { DefaultTasksRunnerOptions } from './default-tasks-runner';
+import { output } from '../utils/output';
+import { getPrintableCommandArgsForTask } from './utils';
+import { Batch } from './tasks-schedule';
+import { stripIndents } from '../utils/strip-indents';
+import { Task } from '../config/task-graph';
+import { Worker } from 'jest-worker';
+import { BatchResults } from './batch/batch-messages';
+import type * as ExecutorWorker from '../../bin/run-executor-worker';
+import { addCommandPrefixIfNeeded } from '../utils/add-command-prefix';
+
+// Worker threads slower for me??? 38s fork, 45s worker thread
+const useWorkerThreads =
+  process.env.NX_JEST_WORKER_TASK_RUNNER_USE_THREADS === 'true';
+
+export class JestWorkerTaskRunner {
+  workspaceRoot = workspaceRoot;
+  cliPath = require.resolve(`../../bin/run-executor-worker.js`);
+
+  private processes = new Set<ChildProcess>();
+
+  private worker = (useWorkerThreads
+    ? new Worker(this.cliPath, {
+        numWorkers: this.options.parallel,
+        enableWorkerThreads: true,
+        exposedMethods: ['executeTask'],
+      })
+    : new Worker(this.cliPath, {
+        numWorkers: this.options.parallel,
+        exposedMethods: ['executeTask'],
+        forkOptions: {
+          env: {
+            ...process.env,
+            ...this.getNxEnvVariablesForForkedProcess(),
+          },
+          // execArgv: ['--inspect-brk'],
+        },
+      })) as Worker & typeof ExecutorWorker;
+
+  constructor(private readonly options: DefaultTasksRunnerOptions) {
+    this.setupOnProcessExitListener();
+  }
+
+  // TODO: vsavkin delegate terminal output printing
+  public forkProcessForBatch({
+    executorName,
+    taskGraph,
+  }: Batch): Promise<BatchResults> {
+    throw new Error('Not implemented');
+  }
+
+  public async executeTask(
+    task: Task,
+    {
+      streamOutput,
+      temporaryOutputPath,
+    }: {
+      streamOutput: boolean;
+      temporaryOutputPath: string;
+    }
+  ): Promise<{ code: number; terminalOutput: string }> {
+    let code: number | undefined;
+    let error: string | undefined;
+    try {
+      if (streamOutput) {
+        output.logCommand(getPrintableCommandArgsForTask(task).join(' '));
+        output.addNewline();
+      }
+
+      const result = await this.worker.executeTask(task, {
+        workspaceRoot: this.workspaceRoot,
+        outputPath: temporaryOutputPath,
+        streamOutput,
+        captureStderr: this.options.captureStderr,
+
+        // Worker threads have trouble serializing these functions but forks don't?
+        onStdout: useWorkerThreads
+          ? undefined
+          : (chunk) => {
+              if (streamOutput) {
```

Comment:
My best attempt at `streamOutput` support since the only stream jest-worker provides is the stream to _all_ the workers. A custom pool implementation could likely improve on it